SIPS/sip-29.md

```diff
@@ -35,85 +35,147 @@ in uppercase in this document are to be interpreted as described in [RFC
 2. Any time an asset needs to be identified, it MUST use the [CAIP-19][caip-19]
 representation.
 
-### Snap Assets API
+### Snap Manifest
+
+This SIP introduces a new permission named `endowment:assets`.
+This permission grants a Snap the ability to provide asset information to the client.
+
+This permission is specified as follows in `snap.manifest.json` files:
+
+```json
+{
+  "initialPermissions": {
+    "endowment:assets": {
+        "chains": [
+            "bip122:000000000019d6689c085ae165831e93"
+        ]
+    }
+  }
+}
+```
+
+`chains` - A non-empty array of CAIP-2 chain IDs that the snap supports. This field is useful for a client in order to avoid unnecessary overhead.
+
+### Snap Implementation
 
 Two methods are defined in the Snap Assets API:
 
-#### Get Token Description
+Any Snap that wishes to provide asset information **MUST** implement the following API:
+
+#### Get Asset Description
+
+```typescript
+import { OnAssetDescriptionHandler } from "@metamask/snaps-sdk";
+
+export const onAssetDescription: OnAssetDescriptionHandler = async ({
```

Comment:
I'm guessing this will be covered in another doc, but I'm curious how data is added to this list provided by the caller. How are assets discovered originally?

Reply:
This is covered by the kering API with the `listAccountAssets` method IIRC @danroc can provide more details about it

Reply:
Then the snap uses any API that provides asset metadata to create its token description